### DISCUSSION TITLE
Too many buildings rendered.

### DISCUSSION THREAD

**Discussion by TenUchikawa** [maplibre/maplibre-gl-js#2001] [Q&A]
Hi I'm Ten.

I am implementing fill-extrusion for the buildings, but it seems that the number of buildings rendered is too much compared to when done with mapbox gl js.

I am creating a mobile app, and it seems to be very taxing on the device to have so many rendered.

Is there any way to render only the buildings near the user's camera like in mapbox?

↓mapBox
![image](https://user-images.githubusercontent.com/63717580/209673032-2fc57688-2131-49ed-80d2-b567c4e5092b.png)

↓maplibre
![image](https://user-images.githubusercontent.com/63717580/209673607-3da72eb6-99cd-4ca7-926d-d7246b8a40f5.png)

**Comment by HarelM** [comment]
You are showing two images with different location and different pitch and with different styles, so it's not a valid comparison.
Generally speaking, I would consider limiting the zoom the building is showing in - this means that further away from the camera the zoom that the map uses is lower and thus will not show the buildings.

**Comment by TenUchikawa** [comment]
I thought about controlling it by zoom level, but the user experience of the mobile app is not very good, so I would like to be able to see the buildings from a lower zoom level (about 14,15).

Is there a feature that would allow me to change the distance at which the buildings are rendered?


Retaked picture
{
  "center": {
      "lng": -73.9177810977589,
      "lat": 40.707373759792944
  },
  "bearing": 18,
  "pitch": 70,
  "zoom": 15
}

mapbox 
![image](https://user-images.githubusercontent.com/63717580/209680450-62724c7a-d018-40c1-be40-eb16b42c8a90.png)

maplibre
![image](https://user-images.githubusercontent.com/63717580/209680538-338fdc8e-054d-424e-9607-a22146bd37f5.png)
